core/src/test/scala/integration/kafka/server/RaftClusterTest.scala

```diff
@@ -212,4 +214,103 @@ class RaftClusterTest {
+            }
+          }
+
+          val (describeResult, ok) = TestUtils.computeUntilTrue(describeOrFail(filter)) {
+            results => results.getOrDefault(entity, java.util.Collections.emptyMap[String, java.lang.Double]()).size() == expectCount
+          }
+          assertTrue(ok, "Broker never saw new client quotas")
+          describeResult
+        }
+
+        var describeResult = alterThenDescribe(entity,
+          Seq(new ClientQuotaAlteration.Op("request_percentage", 0.99)), filter, 1)
+        assertEquals(0.99, describeResult.get(entity).get("request_percentage"), 1e-6)
+
+        describeResult = alterThenDescribe(entity, Seq(
+          new ClientQuotaAlteration.Op("request_percentage", 0.97),
+          new ClientQuotaAlteration.Op("producer_byte_rate", 10000),
+          new ClientQuotaAlteration.Op("consumer_byte_rate", 10001)
+        ), filter, 3)
+        assertEquals(0.97, describeResult.get(entity).get("request_percentage"), 1e-6)
+        assertEquals(10000.0, describeResult.get(entity).get("producer_byte_rate"), 1e-6)
+        assertEquals(10001.0, describeResult.get(entity).get("consumer_byte_rate"), 1e-6)
+
+        describeResult = alterThenDescribe(entity, Seq(
+          new ClientQuotaAlteration.Op("request_percentage", 0.95),
+          new ClientQuotaAlteration.Op("producer_byte_rate", null),
+          new ClientQuotaAlteration.Op("consumer_byte_rate", null)
+        ), filter, 1)
+        assertEquals(0.95, describeResult.get(entity).get("request_percentage"), 1e-6)
+
+        describeResult = alterThenDescribe(entity, Seq(
+          new ClientQuotaAlteration.Op("request_percentage", null)), filter, 0)
+
+        describeResult = alterThenDescribe(entity,
+          Seq(new ClientQuotaAlteration.Op("producer_byte_rate", 9999)), filter, 1)
+        assertEquals(9999.0, describeResult.get(entity).get("producer_byte_rate"), 1e-6)
+
+        // Add another quota for a different entity with same user part
+        val entity2 = new ClientQuotaEntity(Map("user" -> "testkit", "client-id" -> "some-client").asJava)
+        filter = ClientQuotaFilter.containsOnly(
+          List(
+            ClientQuotaFilterComponent.ofEntity("user", "testkit"),
+            ClientQuotaFilterComponent.ofEntity("client-id", "some-client"),
+          ).asJava)
+        describeResult = alterThenDescribe(entity2,
+          Seq(new ClientQuotaAlteration.Op("producer_byte_rate", 9998)), filter, 1)
+        assertEquals(9998.0, describeResult.get(entity2).get("producer_byte_rate"), 1e-6)
+
+        // non-strict match
+        filter = ClientQuotaFilter.contains(
+          List(ClientQuotaFilterComponent.ofEntity("user", "testkit")).asJava)
+
+        val (describeResult2, ok) = TestUtils.computeUntilTrue(admin.describeClientQuotas(filter).entities().get()) {
+          results => results.size() == 2
+        }
+        assertTrue(ok, "Broker never saw two client quotas")
```

Comment:
The "ok" check feels a little clunky here.  Plus if this fails, we don't get any information.

In the past I've just used `TestUtils.retry` plus a mutable variable to get a result + timed assertion check.  Maybe we could come up with a version of that function that returned a value to be a bit more elegant.  In general it would be nice to get the actual exception on a failure.

Reply:
Yea let me clean this up